packages/core/src/browser/status-bar/status-bar.tsx

```diff
@@ -78,12 +79,26 @@ export class StatusBarImpl extends ReactWidget implements StatusBar {
     constructor(
         @inject(CommandService) protected readonly commands: CommandService,
         @inject(LabelParser) protected readonly entryService: LabelParser,
-        @inject(FrontendApplicationStateService) protected readonly applicationStateService: FrontendApplicationStateService
+        @inject(FrontendApplicationStateService) protected readonly applicationStateService: FrontendApplicationStateService,
+        @inject(PreferenceService) protected readonly preferences: PreferenceService,
     ) {
         super();
         delete this.scrollOptions;
         this.id = 'theia-statusBar';
         this.addClass('noselect');
+        // Don't show the status bar until get `workbench.statusBar.visible` preference with `true` value.
```

Comment:
nitpick: the phrasing of the comment is off:

```suggestion
        // Hide the status bar until the `workbench.statusBar.visible` preference returns with a `true` value.
```

The main nitpick was with: "until get `workbench.statusBar.visible` preference".